scipy/stats/tests/test_multicomp.py

```diff
@@ -79,25 +82,111 @@ def test_unbalanced(self):
         ]
         ref = np.array([4.727e-06, 0.022346, 0.97912, 0.99953, 0.86579])
 
-        res = stats.dunnett(*observations, control=control, random_state=rng)
+        res = stats.dunnett(*samples, control=control, random_state=rng)
 
         assert isinstance(res, DunnettResult)
         # last value is problematic
         assert_allclose(res.pvalue[:-1], ref[:-1], atol=0.015)
 
-    def test_allowance(self):
+    @pytest.mark.parametrize(
+        'alternative, allowance, ci_low, ci_high',
+        [
+            ('two-sided', 11, [0, -9, -16], [22, 13, 6]),
+            ('less', 9, [2, -7, -14], [np.nan, np.nan, np.nan]),
+            ('greater', 9, [np.nan, np.nan, np.nan], [20, 11, 4])
+        ]
+    )
+    def test_allowance(self, alternative, allowance, ci_low, ci_high):
+        # Example (a) from Dunnett1995
         rng = np.random.default_rng(189117774084579816190295271136455278291)
-        observations = [
+        samples = [
             [55, 64, 64],
             [55, 49, 52],
             [50, 44, 41]
         ]
         control = [55, 47, 48]
 
-        res = stats.dunnett(*observations, control=control, random_state=rng)
-        allowance = res._allowance(confidence_level=0.95)
-        assert allowance == pytest.approx(11, rel=1)
+        res = stats.dunnett(
+            *samples, control=control, alternative=alternative,
+            random_state=rng
+        )
+        allowance_ = res._allowance(confidence_level=0.95)
+        assert allowance_ == pytest.approx(allowance, rel=1)
```

Comment:
Why not a numpy `assert_allclose`?

Reply:
No reason that was an automatism before when I started and before I needed to compare arrays. I will change for consistency.